src/ng/services/table.js

```diff
@@ -1,3 +1,80 @@
+class lazyProperty {
+	constructor(object, name, create) {
+		Object.defineProperty(object, name, {
+			configurable: true,
+			enumerable: false,
+			get: function () {
+				return this[name] = create(this);
+			},
+			set: function (value) {
+				Object.defineProperty(this, name, {
+					configurable: true,
+					enumerable: false,
+					value: value,
+					writable: true
+				});
+			}
+		});
+	}
+}
+class Empty {
+	column() {
+		return {cells: []};
+	}
+
+	row() {
+		return {cells: []};
+	}
+
+	cell() {
+		return null;
+	}
+}
+class TableDom extends Empty {
+	constructor(element) {
+		super();
+		this.element = element;
+	}
+
+	static get empty() {
+		return new Empty();
+	}
+
+	column(index) {
+		const element = this.element;
+		const rows = element.rows;
+		const cellsCount = rows[0].cells.length;
+		let obj = {};
+		new lazyProperty(obj, "cells", () => {
+			let result = [];
+			if (index >= 0 && index < cellsCount) {
+				for (let i = 0; i < rows.length; i++) {
+					result.push(rows[i].cells[index]);
+				}
+			}
+			return result;
+		});
+		return obj;
+	}
+
+	row(index) {
+		const element = this.element;
+		const rows = element.rows;
+		if (index >= 0 && index < rows.length) {
+			return rows[index];
+		}
+	}
+
+	cell(row, column) {
+		const element = this.element;
+		const rows = element.rows;
+		const cellsCount = rows[0].cells.length;
+		if (row >= 0 && row < rows.length && column >= 0 && column < cellsCount) {
+			return rows[row].cells[column];
+		}
+	}
+}
+
 export default class Table {
 	constructor(markup) {
 		this.markup = markup;
```

Comment:
I wanted to do it, but I think we need it in navigation(to get cell from keyboard)